### ISSUE TITLE
ModifyArgs causes ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by x4e** [SpongePowered/Mixin#410]
I have the following mixin:
```Java
@ModifyArgs(method = "applyEntityCollision", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/Entity;addVelocity(DDD)V"))
private void modifyArgs(Args args, Entity e) {
    	double x = args.get(0);
    	double y = args.get(1);
    	double z = args.get(2);
	AddEntityVelocityEvent event = new AddEntityVelocityEvent(e, x, y, z);
	dispatchEvent(event);
	args.set(0, event.x);
	args.set(1, event.y);
	args.set(2, event.z);
}
```

This causes the following exception:
```
[09:54:34] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
net.minecraft.util.ReportedException: Ticking entity
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:848) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:743) ~[MinecraftServer.class:?]
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:192) ~[IntegratedServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:592) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_241]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 4
	at org.spongepowered.asm.synthetic.args.Args$1.of(Args$1.java) ~[?:?]
	at net.minecraft.entity.Entity.applyEntityCollision(Entity.java:1688) ~[Entity.class:?]
	at net.minecraft.entity.EntityLivingBase.collideWithEntity(EntityLivingBase.java:2706) ~[EntityLivingBase.class:?]
	at net.minecraft.entity.EntityLivingBase.collideWithNearbyEntities(EntityLivingBase.java:2699) ~[EntityLivingBase.class:?]
	at net.minecraft.entity.EntityLivingBase.onLivingUpdate(EntityLivingBase.java:2626) ~[EntityLivingBase.class:?]
	at net.minecraft.entity.EntityLiving.onLivingUpdate(EntityLiving.java:647) ~[EntityLiving.class:?]
	at net.minecraft.entity.monster.EntityMob.onLivingUpdate(EntityMob.java:49) ~[EntityMob.class:?]
	at net.minecraft.entity.monster.AbstractSkeleton.onLivingUpdate(AbstractSkeleton.java:157) ~[AbstractSkeleton.class:?]
	at net.minecraft.entity.EntityLivingBase.onUpdate(EntityLivingBase.java:2405) ~[EntityLivingBase.class:?]
	at net.minecraft.entity.EntityLiving.onUpdate(EntityLiving.java:346) ~[EntityLiving.class:?]
	at net.minecraft.entity.monster.EntityMob.onUpdate(EntityMob.java:57) ~[EntityMob.class:?]
	at net.minecraft.world.World.updateEntityWithOptionalForce(World.java:2186) ~[World.class:?]
	at net.minecraft.world.WorldServer.updateEntityWithOptionalForce(WorldServer.java:871) ~[WorldServer.class:?]
	at net.minecraft.world.World.updateEntity(World.java:2145) ~[World.class:?]
	at net.minecraft.world.World.updateEntities(World.java:1946) ~[World.class:?]
	at net.minecraft.world.WorldServer.updateEntities(WorldServer.java:643) ~[WorldServer.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:842) ~[MinecraftServer.class:?]
	... 4 more
```
It looks like the problem is somewhere here https://github.com/SpongePowered/Mixin/blob/b46bf4cb74f835d22e9d146bd5eb8994ef8a0af3/src/main/java/org/spongepowered/asm/mixin/injection/invoke/arg/ArgsClassGenerator.java#L285

**Comment by LlamaLad7** [comment]
I believe this is fixed as of 0.8.

**Comment by Mumfrey** [comment]
Closing as I believe this is resolved in current versions.
